arch/x86_64/src/intel64/intel64_lowsetup.c

```diff
@@ -64,16 +77,16 @@ void up_lowsetup(void)
 
   /* Setup pointers for accessing Page table and GDT in high address */
 
-  pdpt = &pdpt_low + X86_64_LOAD_OFFSET;
-  pd   = &pd_low   + X86_64_LOAD_OFFSET;
-  pt   = &pt_low   + X86_64_LOAD_OFFSET;
+  pdpt = (uint64_t *)((uint64_t)&pdpt_low + X86_64_LOAD_OFFSET);
+  pd   = (uint64_t *)((uint64_t)&pd_low   + X86_64_LOAD_OFFSET);
+  pt   = (uint64_t *)((uint64_t)&pt_low   + X86_64_LOAD_OFFSET);
 
-  ist64     = (struct ist_s *)((void *)&ist64_low       + X86_64_LOAD_OFFSET);
-  gdt64     = (struct gdt_entry_s *)((void *)&gdt64_low + X86_64_LOAD_OFFSET);
+  ist64     = (struct ist_s *)((uint64_t)&ist64_low       + X86_64_LOAD_OFFSET);
+  gdt64     = (struct gdt_entry_s *)((uint64_t)&gdt64_low + X86_64_LOAD_OFFSET);
```

Comment:
Although this code was not intended to be portable.
I do think, using uintptr_t is a better idea.

I did used uint64_t during my porting, but later I found out that the code's logic might, somehow, be compatible with x86_32.

Reply:
The compiler complains about aliasing, biggest issue here is that you need to cast to not the pointer type first otherwise if thinks if is uint8[] which overflows.

Reply:
```
chip/intel64_lowsetup.c:67:8: warning: assignment to ‘volatile uint64_t *’ {aka ‘volatile long long unsigned int *’} from incompatible pointer type ‘volatile uint8_t *’ {aka ‘volatile unsigned char *’} [-Wincompatible-pointer-types]
   67 |   pdpt = &pdpt_low + X86_64_LOAD_OFFSET;
```

Reply:
@sonicyang I misunderstood what you were saying.  You are right I have update to use the pointer type.